### ISSUE TITLE
Read the version from the nearest package.json relative to working directory

### ISSUE THREAD

**Issue by zamponotiropita** [axelpale/genversion#3]
In `react-native` the use of additional `package.json` inside the folder structure for absolute imports is common. As I want to place `version.js` inside this folder structure the nearest `package.json` found is not my root `package.json`.
It would be beneficial if reading the version from the nearest `package.json` relative to the current working directory could be added

**Comment by axelpale** [comment]
Thanks for the feedback! Let me ensure I understood the issue correctly before attempting to solve it.

You have a folder structure similar to the following:
```
/
  lib/
    ...
    package.json
    version.js
  ...
  package.json
```

And you want the `version` property of `/package.json` to appear in `/lib/version.js`. The problem you faced was that genversion finds and reads `/lib/package.json` instead of `/package.json` regardless of where you called genversion.

Is this correct?

**Comment by zamponotiropita** [comment]
Hi @axelpale and thanks for the very quick reply!
What you described is exactly the issues that I am facing!

**Comment by zamponotiropita** [comment]
Hi @axelpale , any updates on this one?

**Comment by axelpale** [comment]
@zamponotiropita Yeah, sorry for the delay!

I have a proposal which expands the CWD-feature you proposed.

I agree that using the current working directory (CWD) would be handy and something a user would assume. However, I fear that it would increase the likelihood of ambiguous path problems that are difficult to solve quickly. I fear so because programmatic calls such as calls from watch-jobs and cronjobs can have surprising CWDs. I would like the path to be explicitly stated instead of relying on assumption.

Thus, I propose a `--source <path>` flag to be added. The given path would instruct genversion to read the version from this path. If the path is a directory, genversion would search for package.json in this directory. If the path is a file, genversion would attempt to read the version from it as if it were a package.json.

Yet, I perceive an error in my thinking. The example path `lib/version.js` in the docs already depends on CWD. The proposed `--source` path would also allow relative paths such as `.` which equals CWD. Thus, relying on the CWD does not pose more risks.

Still, I think `--source <path>` being more powerful than for example `--use-cwd`. Especially so because `--source .` would essentially force genversion to look the version from CWD.

What do you think? Would the additional parameter `--source <path>` help in your case?

**Comment by zamponotiropita** [comment]
Hi @axelpale I believe `--source <path>` would be a perfect solution for my case!
Thanks and looking forward to the update!

**Comment by axelpale** [comment]
Thanks! Here you go. Fixed in v2.2.0: https://github.com/axelpale/genversion/commit/bb5d3c445e4a42c7f5473ce622b7a800aea84c08

Here are some design notes for later development.
- I thought of naming the parameter `pjson` instead of `source` but kept `source` because symmetry to `target` and because the given path probably conveys the meaning.
- I thought of adding the source path to the generated version file as a comment but refrained because only absolute paths would be informative but they would put private dir paths into code to commit, which can be a security issue.

**Comment by zamponotiropita** [comment]
Great!
